src/basic/lock-util.c

```diff
@@ -149,6 +149,11 @@ static int fcntl_lock(int fd, int operation, bool ofd) {
         if ((operation & LOCK_NB) && r == -EACCES)
                 r = -EAGAIN;
 
+        /* fcntl() returns EINVAL if the operation is not known to the kernel so let's treat that as a not
+         * supported error. */
```

Comment:
what is this about? really old kernels without unposix locks? iirc they were added in 3.15 which happens to be our minimum baseline?